app/controllers/users_controller.rb

```diff
@@ -20,10 +20,25 @@ def create
     end
   end
 
+  def edit
+    @user = User.find(params[:id])
+  end
+
   def show
     @user = User.find(params[:id])
   end
 
+  def update
+    @user = User.find(params[:id])
+    if @user.update_attributes(update_params)
+      flash[:success] = "Profile updated"
+      redirect_to user_path(@user)
+    else
+      flash.now[:error] = "Invalid Profile Edit. Try Again"
+      render "edit"
```

Comment:
It needs to be render because I want to render the edit page again with the params that hit the update action which will include the mistaken parameter that the user typed in. That way the user can see what he did wrong. Render and redirect_to are subtly different in that way. Also assert template doens't work unless you use render in the controller.